### PR TITLE
Make it work

### PR DESCRIPTION
**Pull request by cjdenio**
This PR makes Hackagotchi great again.

- no more dependency hell (specifically, I pinned rocket to a specific commit and upgraded everything)
- docker-based dev environment
- print statements that I'm too lazy to remove

### CODE REVIEW

**Review by Muirrum — CHANGES_REQUESTED** [review]
Needs `fmt`

**Review by Muirrum — CHANGES_REQUESTED** [review]
Config should not be committed here - we should maintain this separately

File: src/banker.rs [hunk 1]
```diff
@@ -49,6 +49,8 @@ pub async fn message(msg: String) -> Result<(), String> {
 }
 
 pub async fn invoice(user: &str, amount: u64, reason: &str) -> Result<(), String> {
+    println!("Invoicing user {} for {} gp", user, amount);
```
Comment (Muirrum) [inline src/banker.rs:52]: This should be a debug! or info! macro, not println!

File: src/main.rs [hunk 2]
```diff
@@ -1,25 +1,20 @@
-#![warn(missing_docs)]
+#![allow(missing_docs)]
```
Comment (Muirrum) [inline src/main.rs:1]: This was added for a reason

File: src/main.rs [hunk 3]
```diff
@@ -1027,7 +1022,7 @@ fn hackstead_blocks(
             .as_ref()
             .filter(|cert| {
                 if cert.requires_xp {
-                    tiles_owned < hs_adv_sum.land.try_into().unwrap()
+                    tiles_owned < hs_adv_sum.land as usize
```
Comment (Muirrum) [inline src/main.rs:1025]: This still needs error handling, no?

**Review by Muirrum — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 2 above)
Comment (cjdenio) [reply src/main.rs:1]: oops sorry, that was supposed to be temoprary

File: src/main.rs (hunk 3 above)
Comment (cjdenio) [reply src/main.rs:1025]: The original line fails to build with this error, and I can't figure out where to put the type annotations:
```
error[E0283]: type annotations needed
    --> src/main.rs:1025:33
     |
1025 |                     tiles_owned < hs_adv_sum.land.try_into().unwrap()
     |                                 ^ -------------------------- this method call resolves to `Result<T, <Self as std::convert::TryInto<T>>::Error>`
     |                                 |
     |                                 cannot infer type
     |
     = note: cannot satisfy `usize: PartialOrd<_>`
```

File: src/banker.rs [hunk 4]
```diff
@@ -49,8 +49,6 @@ pub async fn message(msg: String) -> Result<(), String> {
 }
 
 pub async fn invoice(user: &str, amount: u64, reason: &str) -> Result<(), String> {
-    println!("Invoicing user {} for {} gp", user, amount);
```
Comment (Muirrum) [inline src/banker.rs]: That wasn't to say that this shouldn't be logged - just not as a `println!` macro

Comment (cjdenio) [reply src/banker.rs]: I thought I'd remove it anyway; I'll be rewriting this module in a bit anyway

**Review by Muirrum — APPROVED** [review]

**Pull request merged**